### DISCUSSION TITLE
Script setup runs twice in nested children and parent page runs every time too.

### DISCUSSION THREAD

**Discussion by vanling** [nuxt/nuxt#17952] [Questions] [answered]
Using multiple nested page levels script setup shows some some weird behaviour. 
- Not sure if parent setup should always be re-executed on every change of child page route. 
- But I am sure the double executed setups in children should not happen right?

All looks ok on server console on a direct page access, but routing in the browser gives double console logs.

<img width="589" alt="Schermafbeelding 2021-12-30 om 12 12 33" src="https://user-images.githubusercontent.com/50699657/147746900-3db07bca-b7f5-476b-9af6-16dd3b6b862d.png">

Let me know if you want me to create an issue for this and/or a reproduction repo.

**issues that might be related**
 - #2194 
 - #2309 
 

**pages/projects.vue**
```vue.js
<template>
  <div>
    <nuxt-nested-page />
  </div>
</template>
<script setup>
  console.log('Project nested, executes one time but again on every child route', new Date().getMilliseconds())
</script>
```
**pages/projects/dashboard.vue**
```vue.js
<template>
  <div>
      some Menu
      <nuxt-nested-page />
  </div>
</template>
<script setup>
  console.log('Gets executed 2 times and on every child page', new Date().getMilliseconds())
</script>
```

**pages/projects/dashboard/index.vue**
```vue.js
<template>
  <div>
      some Menu
      <nuxt-nested-page />
  </div>
</template>
<script setup>
  console.log('Gets called 2 times', new Date().getMilliseconds())
</script>
```

**Comment by gabjauf** [comment]
Same issue here, is there a fix or workaround for this?

**Reply by danielroe** [reply]
This has been resolved. https://stackblitz.com/edit/github-pivfbt?file=app.vue&terminal=dev

If you are experiencing any issues, first check the [breaking changes tracker](https://github.com/nuxt/framework/discussions/2883), and then raise an issue with a reproduction. 🙏

**Comment by vanling** [comment, marked as answer]
> This has been resolved. https://stackblitz.com/edit/github-pivfbt?file=app.vue&terminal=dev
> 
> If you are experiencing any issues, first check the [breaking changes tracker](https://github.com/nuxt/framework/discussions/2883), and then raise an issue with a reproduction. 🙏

marking as answered via: @danielroe
